File: container/aa-week19-docker-compose/proj3/pushing-to-heroku/backend/db/models/item.js
"use strict";
module.exports = (sequelize, DataTypes) => {
  const Item = sequelize.define(
    "Item",
    {
      happiness: DataTypes.INTEGER,
      imageUrl: {
        allowNull: false,
        type: DataTypes.STRING,
        validates: {
          isUrl: true,
          len: [0, 255],
        },
      },
      name: {
        allowNull: false,
        type: DataTypes.STRING,
        validates: {
          len: [0, 255],
        },
      },
      price: {
        allowNull: false,
        type: DataTypes.INTEGER,
      },
      pokemonId: {
        allowNull: false,
        type: DataTypes.INTEGER,
      },
    },
    {}
  );
  Item.associate = function (models) {
    Item.belongsTo(models.Pokemon, { foreignKey: "pokemonId", as: "pokemon" });
  };
  return Item;
};

File: container2/appacademy/dayswork/W9D5/pokedex_pt1/app/models/item.rb
# == Schema Information
#
# Table name: items
#
#  id         :bigint           not null, primary key
#  pokemon_id :integer          not null
#  name       :string           not null
#  price      :integer          not null
#  happiness  :integer          not null
#  image_url  :string           not null
#  created_at :datetime         not null
#  updated_at :datetime         not null
#

class Item < ApplicationRecord
  validates :happiness, :image_url, :name, :price, presence: true
  validates :happiness, :price, numericality: true

  belongs_to :pokemon
end
